drivers/digital-io/max22200/iio_max22200.c

```diff
@@ -0,0 +1,1090 @@
+	uint32_t val;
+	int ret;
+
+	ret = max22200_get_ch_trig(desc->max22200_desc, channel->address,
+				   &ch_trig);
+	if (ret)
+		return ret;
+
+	val = (uint32_t)ch_trig;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, (int32_t *)&val);
+}
+
+/**
+ * @brief Read the channel_drive/channel_side/channel_op_mode attribute.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+*/
+static int max22200_iio_read_ch_mode(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_ch_drive ch_drive;
+	enum max22200_ch_side ch_side;
+	enum max22200_ch_op_mode ch_op_mode;
+	int ret;
+
+	ret = max22200_get_ch_mode(desc->max22200_desc, channel->address,
+				   &ch_drive, &ch_side, &ch_op_mode);
+	if (ret)
+		return ret;
+
+	switch (priv) {
+	case MAX22200_IIO_CH_DRIVE:
+		return sprintf(buf, "%s ", max22200_ch_drive_avail[ch_drive]);
+	case MAX22200_IIO_CH_SIDE:
+		return sprintf(buf, "%s ", max22200_ch_side_avail[ch_side]);
+	case MAX22200_IIO_CH_OP_MODE:
+		return sprintf(buf, "%s ",
+			       max22200_ch_op_mode_avail[ch_op_mode]);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Read the channel_freq attribute for a specific channel
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_read_channel_freq(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_ch_freq ch_freq;
+	int ret;
+
+	ret = max22200_get_ch_freq(desc->max22200_desc, channel->address,
+				   &ch_freq);
+	if (ret)
+		return ret;
+
+	return sprintf(buf, "%s ", max22200_channel_freq_avail[ch_freq]);
+}
+
+/**
+ * @brief Read the enable attribute for a specific channel
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_read_enable(void *dev, char *buf, uint32_t len,
+				    const struct iio_ch_info *channel,
+				    intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	bool val = 0;
+	int ret;
+
+	ret = max22200_get_ch_enable(desc->max22200_desc, channel->address,
+				     priv, &val);
+	if (ret)
+		return ret;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1,
```

Comment:
casting from bool to int32_t is not a good idea. please make the update also for this case

Reply:
Using different approach then.